File: neet_code_all_cpp/stack/removing_stars_from_string.cpp
#include <iostream>
#include <string>

// 模範解答(Python)は全く同じだった
class Solution {
    public:
        // AC
        std::string removingStars(std::string s) {
            std::string result;
            for(auto& c: s) {
                if(c == '*') {
                    result.pop_back();
                } else {
                    result.push_back(c);
                }
            }

            return result;
        }
};

int main(void) {
    std::string case_1 = "leet**cod*e";
    std::string case_2 = "erase*****";

    Solution s_1;

    std::cout << s_1.removingStars(case_1) << std::endl;
    std::cout << s_1.removingStars(case_2) << std::endl;
}

File: neet_code_all/src/stack/removing_stars_from_string.rs
// 模範解答(Python)は全く同じだった
struct Solution;
impl Solution {
    // AC
    pub fn remove_stars(s: String) -> String {
        let mut result = vec![];
        for c in s.chars() {
            if c == '*' {
                result.pop();
            } else {
                result.push(c);
            }
        }

        result.iter().collect::<String>()
    }
}

fn main() {
    let case_1 = "leet**cod*e".to_string();
    // => "lecoe"
    let case_2 = "erase*****".to_string();
    // => ""

    println!("case_1: {}", Solution::remove_stars(case_1.clone()));
    println!("case_2: {}", Solution::remove_stars(case_2.clone()));
}
